src/tsolvers/egraph/EnodeStore.h

```diff
@@ -35,31 +35,54 @@ class Logic;
 struct PTRefERefPair { PTRef tr; ERef er; };
 
 class EnodeStore {
+
+    struct SignatureHash {
+        EnodeAllocator const & ea;
+
+        SignatureHash(EnodeAllocator const & ea) : ea{ea} {}
+
+        std::size_t operator()(ERef ref) const {
+            Enode const & node = ea[ref];
+            std::size_t seed = node.getSymbol().x;
+            for(uint32_t i = 0; i < node.getSize(); ++i) {
+                seed ^= ea[node[i]].getRoot().x + 0x9e3779b9 + (seed << 6) + (seed >> 2);
```

Comment:
How did you come up with this hash function?  Is it a concern that (I assume) most of the time the most significant bits of `x` will be zero?  I don't really have an alternate suggestion at the moment, but just asking out of curiosity and maybe documentation purposes.

Reply:
Ok thanks!

Reply:
Added comment in the code.